### PR TITLE
Added ediff1d to numpy frontend

### LINKED ISSUE #13356: ediff1d

### PR DESCRIPTION
**Pull request by Aneesh02**
Close #13356

### CODE REVIEW

**Comment by Aneesh02** [comment]
@Infrared1029 Request you to look at my PR

**Comment by Infrared1029** [comment]
Hello @Aneesh02, the test appears to be failing, can you make sure it passes so I can take a close look? Thanks!

**Comment by Aneesh02** [comment]
Hello @Infrared1029, Thanks for reviewing I have fixed the error in test_run_1. The linting error is not from the files I have changed.

**Comment by Infrared1029** [comment]
Hello @Aneesh02, the implementation looks good to me, just a few comments, make sure to follow the original function signature i.e `numpy.ediff1d(ary, to_end=None, to_begin=None)` instead of `numpy.ediff1d(x, to_end=None, to_begin=None)`, the test appears to generate values for `x` only, this is sub-optimal, the test has to generate values for all possible inputs the function takes, i.e you should generate values for `to_begin` and `to_end` as well, the tests fails with `paddle` backend, it has to do with the dtypes (paddle.diff does not support uint8), you might need to add a `with_unsupported_dtypes` decorator to the backend `diff` implementation for paddlepaddle. Thanks!

**Comment by Aneesh02** [comment]
@Infrared1029 Thanks for the review. I have incorporated the original function signature. Can you please help me as to where can I find paddle.diff in the backend files?

**Comment by Aneesh02** [comment]
@Infrared1029 please help me resolve this error

Screenshot from 2023-04-02 12-20-03

**Comment by Aneesh02** [comment]
UPDATE: All the tests are passing for now. Just the test file needs to fixed to incorporate values for `to_begin` and `to_end`.

**Comment by Aneesh02** [comment]
@Infrared1029 I have fixed the issues. Please have a look. Thanks!

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py [hunk 1]
```diff
@@ -329,3 +329,36 @@ def test_numpy_diff(
...
     )
+    
+    
+# ediff1d
+@handle_frontend_test(
+    fn_tree="numpy.ediff1d",
+    dtype_x_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        min_num_dims=2,
+        max_num_dims=2,
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py:340]: The number of dimensions should not always be equal to 2. We ought to test for all cases.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py [hunk 2]
```diff
@@ -329,3 +329,36 @@ def test_numpy_diff(
         x=x[0],
         axis=axis,
     )
+    
+    
+# ediff1d
+@handle_frontend_test(
+    fn_tree="numpy.ediff1d",
+    dtype_x_axis=helpers.dtype_values_axis(
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py:337]: You are not using axis anywhere, so I would suggest using `helpers.dtype_and_values`.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py [hunk 3]
```diff
@@ -329,3 +329,36 @@ def test_numpy_diff(
...
+        available_dtypes=helpers.get_dtypes("numeric"),
+        min_num_dims=2,
+        max_num_dims=2,
+        min_dim_size=1,
+        max_dim_size=10,
+        valid_axis=True,
+        force_int_axis=True,
+    ),
+)
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py:346]: You will need to generate `to_begin` and `to_end`. These can be either a single number or an array.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py [hunk 4]
```diff
@@ -329,3 +329,31 @@ def test_numpy_diff(
...
         axis=axis,
     )
+    
+    
+# ediff1d
+@handle_frontend_test(
+    fn_tree="numpy.ediff1d",
+    dtype_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        num_arrays=3
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py:339]: Based on the error I see in the CI test, I think you'll need to set the minimum number of dimensions to 1.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py (hunk 4 above)
Comment (Aneesh02) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py:339]: Thank you for checking it out. Appreciate it.

Comment (Aneesh02) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py]: Also min_dim_size should be kept as 1

**Comment by Aneesh02** [comment]
@AnnaTz @Infrared1029 I have tried a lot I cannot find out why the error is coming. Please help me resolve it. The example generated is fine.

**Comment by AnnaTz** [comment]
> @AnnaTz @Infrared1029 I have tried a lot I cannot find out why the error is coming. Please help me resolve it. The example generated is fine.

Just found ediff1 in our jax numpy frontend. You could take a look at how that was implemented and tested. It's practically the same function.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py [hunk 5]
```diff
@@ -329,3 +329,43 @@ def test_numpy_diff(
...
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"), 
+        min_num_dims=1, 
+        max_num_dims=1
+    ),
+    to_end=helpers.ints(
+        min_value=-1,
+        max_value=10,
+    ),
+    to_begin=helpers.ints(
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py:346]: To be totally correct, `to_end` and `to_begin` should be either an integer (which you've done) or a sequence. You could use `st.one_of` like so `st.one_of(st.integers(-1, 10), st.lists(st.integers(-1, 10), min_size=1, max_size=10))`.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

**Comment by Aneesh02** [comment]
Thanks @AnnaTz @Infrared1029

**Pull request merged**